### ISSUE TITLE
pylint.exceptions.UnknownMessageError: No such message id or symbol 'nonstandard-exception'.

### ISSUE THREAD

**Issue by gangefors** [PyCQA/pylint#2931]
### Steps to reproduce
Run pylint using python 3.7 on a file with the following content.
```
"""Test nonstandard-exception crash in python 3.7"""
from http import HTTPStatus
from typing import Callable

import requests


# The HTTP status code that can reasonably be argued to be transient.
# Status codes that aren't actually used (e.g. WebDAV ones) weren't
# considered for inclusion.
TRANSIENT_HTTP_ERRORS = (
    HTTPStatus.REQUEST_TIMEOUT,
    HTTPStatus.TOO_MANY_REQUESTS,
    HTTPStatus.INTERNAL_SERVER_ERROR,
    HTTPStatus.BAD_GATEWAY,
    HTTPStatus.SERVICE_UNAVAILABLE,
    HTTPStatus.GATEWAY_TIMEOUT,
    HTTPStatus.INSUFFICIENT_STORAGE,
)


class TransientError(Exception):
    """Indicates an error that can be expected to be transient,
    i.e. it will hopefully disappear with repeat attempts at
    the same operation.
    """


def translate_exceptions(function: Callable) -> Callable:
    """Wrap a callable in exception handlers that translate exceptions
    that are known to be transient into TransientError exceptions,
    making it easier to distinguish transient errors from permanent
    ones (for retry purposes).

    If an instance of an exception class is translated to another
    class, i.e. the exception is caught and reraised with another
    type, the __context__ member of the reraised exception will
    contain the original exception.

    Specifically, transient instances of the following exception classes
    are translated:
        - requests.HTTPError

    Args:
        function: The function to wrap.

    Returns:
        A callable that wraps `function` but with additional exception traps.

    Raises:
        TransientError: If the function called raises an exception that's
            deemed to be of a transient nature.
    """
    def _wrapper(*args, **kwargs):
        try:
            return function(*args, **kwargs)
        except requests.HTTPError as err:
            if err.response.status_code in TRANSIENT_HTTP_ERRORS:
                raise TransientError(str(err))
            raise
    return _wrapper


def do_stuff():
    """ Get and parse the team test configuration """
    try:
        pass
    except TransientError as t_error:
        # Not so transient after all, raise the original cause.
        raise t_error.__context__
```

### Current behavior
Pylint crashes with the following error.
```
Traceback (most recent call last):
  File "/home/stefanga/.pyenv/versions/3.7.3/bin/pylint", line 10, in <module>
    sys.exit(run_pylint())
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/__init__.py", line 20, in run_pylint
    Run(sys.argv[1:])
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/lint.py", line 1628, in __init__
    linter.check(args)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/lint.py", line 943, in check
    self._do_check(files_or_modules)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/lint.py", line 1075, in _do_check
    self.check_astroid_module(ast_node, walker, rawcheckers, tokencheckers)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/lint.py", line 1158, in check_astroid_module
    walker.walk(ast_node)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/utils.py", line 1303, in walk
    self.walk(child)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/utils.py", line 1303, in walk
    self.walk(child)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/utils.py", line 1303, in walk
    self.walk(child)
  [Previous line repeated 1 more time]
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/utils.py", line 1300, in walk
    cb(astroid)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/checkers/exceptions.py", line 319, in visit_raise
    ExceptionRaiseLeafVisitor(self, node).visit(inferred_value)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/checkers/exceptions.py", line 184, in visit
    dispatch_meth(node)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/checkers/exceptions.py", line 235, in visit_classdef
    self._checker.add_message("nonstandard-exception", node=self._node)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/checkers/__init__.py", line 110, in add_message
    self.linter.add_message(msg_id, line, node, args, confidence, col_offset)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/utils.py", line 493, in add_message
    message_definitions = self.msgs_store.get_message_definitions(msg_descr)
  File "/home/stefanga/.pyenv/versions/3.7.3/lib/python3.7/site-packages/pylint/utils.py", line 1021, in get_message_definitions
    raise UnknownMessageError(error_msg)
pylint.exceptions.UnknownMessageError: No such message id or symbol 'nonstandard-exception'.
```

### Expected behavior
pylint doesn't crash. It should either handles the code correclty or flag the code with bad-exception-context which can be disabled.

### pylint --version output
```
pylint 2.3.1
astroid 2.2.5
Python 3.7.3 (default, May  6 2019, 17:40:30) 
[GCC 6.3.0 20170516]
```

**Comment by PCManticore** [comment]
Thank you, we should definitely change this behaviour.

**Comment by sameenislam** [comment]
I'm encountering a similar issue and `pylint --version` outputs:

````
pylint 2.16.2
astroid 2.14.2
Python 3.10.9 | packaged by conda-forge | (main, Feb  2 2023, 20:24:27) [Clang 14.0.6 ]
````

Stacktrace:

````
2023-02-23 16:55:41.609 [info] [Error - 16:55:41] Linting failed with error:
Traceback (most recent call last):
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/tool/lsp_server.py", line 106, in _linting_helper
    result = _run_tool_on_document(document, use_stdin=True, extra_args=extra_args)
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/tool/lsp_server.py", line 573, in _run_tool_on_document
    result = utils.run_module(
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/tool/lsp_utils.py", line 156, in run_module
    return _run_module(module, argv, use_stdin, source)
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/tool/lsp_utils.py", line 141, in _run_module
    runpy.run_module(module, run_username="__main__")
  File "/Users/username/miniconda3/envs/username-deploy/lib/python3.10/runpy.py", line 227, in run_module
    return _run_code(code, {}, init_globals, run_username, mod_spec)
  File "/Users/username/miniconda3/envs/username-deploy/lib/python3.10/runpy.py", line 86, in _run_code
    exec(code, run_globals)
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/libs/pylint/__main__.py", line 10, in <module>
    pylint.run_pylint()
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/libs/pylint/__init__.py", line 36, in run_pylint
    PylintRun(argv or sys.argv[1:])
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/libs/pylint/lint/run.py", line 160, in __init__
    linter.enable("c-extension-no-member")
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/libs/pylint/lint/message_state_handler.py", line 224, in enable
    self._set_msg_status(
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/libs/pylint/lint/message_state_handler.py", line 148, in _set_msg_status
    message_definitions = self._get_messages_to_set(msgid, enable, ignore_unknown)
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/libs/pylint/lint/message_state_handler.py", line 131, in _get_messages_to_set
    message_definitions = self.linter.msgs_store.get_message_definitions(msgid)
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/libs/pylint/message/message_definition_store.py", line 74, in get_message_definitions
    for m in self.message_id_store.get_active_msgids(msgid_or_symbol)
  File "/Users/username/.vscode/extensions/ms-python.pylint-2023.2.0/bundled/libs/pylint/message/message_id_store.py", line 159, in get_active_msgids
    raise UnknownMessageError(error_msg)
pylint.exceptions.UnknownMessageError: No such message id or symbol 'c-extension-no-member'.
````

Are there any known fixes?

**Comment by sameenislam** [comment]
Ok, I was able to solve this issue by uninstalling the `conda-forge` version of `pylint`. I then reinstalled `pylint` using `pip install pylint` and reloaded my VSCode window. Now the linting is working.
